sdk/servicebus/service-bus/src/core/receiverHelper.ts

```diff
@@ -37,9 +37,10 @@ export class ReceiverHelper {
   }
 
   /**
-   * Prevents us from receiving any further messages.
+   * Drains the receiver, preventing any new activity by
+   * causing addCredits to no longer add any credits.
    */
-  public async stopReceivingMessages(): Promise<void> {
+  public async suspend(): Promise<void> {
     const receiver = this._getCurrentReceiver();
 
     if (receiver == null) {
```

Comment:
What about the case where the receiver is not null, but is not open as well? Or is in the process of getting recovered in the `onDetached()` call? 

In the first case, we would not need drain credits. I doubt that we will get the `receiver_drained` event at all in such cases. Same goes for the case when there is a problem with the connection.
In the second, we should stop the recovery process.

Reply:
Great points. I'll add some tests in.

Reply:
Made some modifications so we check out the receiver and make sure it's open before doing anything to it. For the most part this makes ReceiverHelper's methods no-ops.

There are some new tests in receiverHelpers.spec.ts for it.